modules/network/src/test/java/org/apache/ignite/internal/network/serialization/MarshallableTest.java

```diff
@@ -0,0 +1,173 @@
+                } catch (Exception e) {
+                    throw new RuntimeException(e);
+                }
+            }
+
+            @Override
+            public <T> SerializationResult write(T object) {
+                try (ByteArrayOutputStream baos = new ByteArrayOutputStream(); ObjectOutputStream oos = new ObjectOutputStream(baos)) {
+                    oos.writeObject(object);
+                    oos.close();
+                    return new SerializationResult(baos.toByteArray(), Collections.singletonList(descriptor.descriptorId()));
+                } catch (Exception e) {
+                    throw new RuntimeException(e);
+                }
+            }
+
+            @Override
+            public ClassDescriptor getClassDescriptor(int typeDescriptorId) {
+                return descriptorContext.getDescriptor(typeDescriptorId);
+            }
+
+            @Override
+            public ClassDescriptor getClassDescriptor(String typeName) {
+                assertEquals(descriptor.className(), typeName);
+
+                return descriptor;
+            }
+
+            @Override
+            public boolean isBuiltin(int typeDescriptorId) {
+                return ClassDescriptorFactoryContext.isBuiltin(typeDescriptorId);
+            }
+        };
+
+        var serializationService = new SerializationService(registry, userObjectSerializer);
+        var perSessionSerializationService = new PerSessionSerializationService(serializationService);
+        final var decoder = new InboundDecoder(perSessionSerializationService);
+
+        // List that holds decoded object
+        final var list = new ArrayList<>();
+
+        var writer = new DirectMessageWriter(perSessionSerializationService, ConnectionManager.DIRECT_PROTOCOL_VERSION);
+
+        // Test map that will be sent as a Marshallable object within the MessageWithMarshallable message
+        Map<String, SimpleSerializableObject> testMap = Map.of("test", new SimpleSerializableObject(10));
+
+        MessageWithMarshallable msg = msgFactory.messageWithMarshallable().marshallableMap(testMap).build();
+
+        MessageSerializer<NetworkMessage> serializer = registry.createSerializer(msg.groupType(), msg.messageType());
+
+        ByteBuffer nioBuffer = ByteBuffer.allocate(10_000);
+
+        writer.setBuffer(nioBuffer);
+
+        // Write a message to the ByteBuffer.
+        boolean fullyWritten = serializer.writeMessage(msg, writer);
+
+        assertTrue(fullyWritten);
+
+        int size = nioBuffer.position();
+
+        nioBuffer.flip();
+
+        ByteBuf buffer = allocator.buffer();
+
+        for (int i = 0; i < size; i++) {
+            // Write bytes to a decoding buffer one by one
+            buffer.writeByte(nioBuffer.get());
+
+            decoder.decode(ctx, buffer, list);
+
+            if (i < size - 1) {
+                // Any time before the buffer is fully read, message object should not be decoded
+                assertEquals(0, list.size());
```

Comment:
This can be replaced with (arguably more readable)

```
assertThat(list, is(empty()));
```

Plain English prose instead of a techy expression again :)